retrieval/discovery/dns.go

```diff
@@ -25,12 +25,18 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 
 	clientmodel "github.com/prometheus/client_golang/model"
-
 	"github.com/prometheus/prometheus/config"
-	"github.com/prometheus/prometheus/utility"
 )
 
-const resolvConf = "/etc/resolv.conf"
+const (
+	resolvConf = "/etc/resolv.conf"
+
+	DNSSourcePrefix = "dns"
```

Comment:
I've had the situation of "I wish this was exported" way more often than "I wish this was unexported". As `TargetProvider` and `TargetGroups` reveal source identifiers, those constants might be of interest.
Certainly low risk to export some constants.